scripts/prepare_sector_network.py

```diff
@@ -2161,8 +2166,8 @@ def add_rail_transport(n, costs):
     n = pypsa.Network(snakemake.input.network, override_component_attrs=overrides)
 
     # Fetch the coutry list from the network
-    countries = list(n.buses.country.unique())
-
+    # countries = list(n.buses.country.unique())
+    countries = snakemake.config["countries"]

```

Comment:
What is the advantage fetching the countries from the config? Isn't it more failsafe to get it from the network?

Reply:
Yes and it also avoids breaking the workflow in cases like this: https://github.com/pypsa-meets-earth/pypsa-earth/issues/649